enos/tests/utils/test_provider.py

```diff
@@ -2,6 +2,35 @@
 from enos.utils.provider import *
 import copy
 
+class TestLoadConfig(unittest.TestCase):
+    def test_load_config_with_topology(self):
+        config = {
+                'topology':{
+                    'grp1': {
+                        'a':{
+                            'control': 1,
+                            }
+                        },
+                    'grp[2-6]': {
+                        'a':{
+                            'compute': 10,
+                            }
+                        }
+                    },
+                'provider': 'test_provider'
```

Comment:
I wasn't sure if we enfore a default provider. 
I think the config dict must have a provider key